ui/chapter/Chapter.tsx

```diff
@@ -123,15 +131,17 @@ export default function Chapter({ children, metadata, lang }) {
                       (chapter.metadata.lessons.length === 0 && null)}
                     <div className="flex pt-8 md:w-full">
                       <Button
-                        href={`${routes.chaptersUrl}/${chapter.metadata.slug}/${chapter.metadata.intros[0]}`}
+                        href={`${routes.chaptersUrl}/${chapter.metadata.slug}/${linkFrag}`}
```

Comment:
using something along the lines of
``` suggestion
                        href={`${keysMeta.progress.path}`}
```

or something similar might simplify the logic

**I noticed that the `keysMeta` already has the `chapters` in the url what do you think about removing it in a future PR as we already have this `routes.chapters` logic built in

Reply:
> I noticed that the keysMeta already has the chapters in the url what do you think about removing it in a future PR as we already have this routes.chapters logic built in

This would be a fair simplification while also introducing consistency in the manner we use hrefs in the codebase.

Reply:
I opened a PR #478 so we remember